RevenueCatUI/Templates/Components/Packages/PurchaseButton/PurchaseButtonComponentView.swift

```diff
@@ -20,15 +20,32 @@ import SwiftUI
 @available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
 struct PurchaseButtonComponentView: View {
 
+    @EnvironmentObject
+    private var paywallState: PaywallState
+
+    @EnvironmentObject
+    private var purchaseHandler: PurchaseHandler
+
     private let viewModel: PurchaseButtonComponentViewModel
 
     internal init(viewModel: PurchaseButtonComponentViewModel) {
         self.viewModel = viewModel
     }
 
     var body: some View {
-        Button {
-            // WIP: Need to perform purchase logic
+        AsyncButton {
+            guard !self.purchaseHandler.actionInProgress else { return }
+
+            // WIP: Need to log warning if currently subscribed
+            guard let selectedPackage = self.paywallState.selectedPackage
+//                    , selectedPackage.currentlySubscribed
+            else {
+                Logger.warning(Strings.product_already_subscribed)
+                return
+            }
```

Comment:
This is logic that was implemented in a very different way in Paywalls V1. This will be done in a new further PR 💪